docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml

```diff
@@ -309,16 +312,71 @@
+      shell: source /etc/platform/openrc; system application-list | grep deployment-manager | awk '{ print $10 }'
+      register: dm_app_status
 
-    # Install or Upgrade DM-Monitor
-    - name: "Install or Upgrade DM-monitor"
-      shell: "ansible-playbook {{ dm_monitor_playbook }}"
-      become: yes
-      ignore_errors: yes
-      when: dm_monitor_playbook
+    - name: Check if unlock is ready
+      stat:
+        path: /etc/platform/.unlock_ready
+      register: is_unlock_ready
+
+    # DM system app check, upload/apply helm-chart
+    - block:  
+        # Find dm-system app helm-chart
+       - name: Find DM system-app helm-chart in application helm-chart directory
+         find:
+          paths: /usr/local/share/applications/helm/
+          patterns: 'deployment-manager*.tgz'
+         register: app_helm_chart
+
+       - name: Fail DM system app helm-chart does not exists
+         fail:
+           msg: "DM system app helm-chart does not exists in application directory"
+         when: (app_helm_chart.files[0].path | length == 0)
+   
+       - block:
+          # dm-system app upload/apply
+         - name: Upload Deployment Manager System App Helm Chart
+           shell: source /etc/platform/openrc; system application-upload {{ manager_app_chart }}
+
+         - wait_for:
+           # Pause for an arbitrary amount of time to allow the deployment
+           # manager app to be uploaded.
+             timeout: 6
+             msg: Waiting for the Deployment Manager App to be uploaded
+      
+         - name: Search Deployment Manager Upload Status
+           shell: source /etc/platform/openrc; system application-list | grep deployment-manager | awk '{ print $10 }'
+           register: app_upload_status
+
+         - name: Fail DM system app upload fails
+           fail:
+             msg: "The upload of DM system app has failed."
+           when: ("upload-failed" in app_upload_status.stdout)
+
+         - name: Apply Deployment Manager System App
+           shell: source /etc/platform/openrc; system application-apply deployment-manager   
+           when: ("uploaded" in app_upload_status.stdout)
+
+         - wait_for:
+           # Pause for an arbitrary amount of time to allow the deployment
+           # manager app to be applied.
+             timeout: 80
+             msg: Waiting for the Deployment Manager App to be applied
+
+         - name: Search Deployment Manager Apply Status
+           shell: source /etc/platform/openrc; system application-list | grep deployment-manager | awk '{ print $10 }'
+           register: app_apply_status
+
+         - name: Fail DM system app apply fails
+           fail:
+             msg: "The apply of DM system app has failed."
+           when: ("apply-failed" in app_apply_status.stdout)
+
+         when: (app_helm_chart.files[0].path | length != 0)
+
+      when: ((dm_app_status.stdout | length == 0) and (is_unlock_ready.stat.exists == True))
```

Comment:
What is the "is_unlock_ready" mean here? Why do we want to check it for this whole block? Can you add some comments above the block or here to describe that?